torch/export/exported_program.py

```diff
@@ -681,6 +681,28 @@ def _decompose_exported_program(
     return exported_program
 
 
+@contextmanager
+def _disabled_load_state_dict_hooks(mod: torch.nn.Module):
+    state_dict_hooks: Dict[int, Callable] = dict(mod._state_dict_hooks)
+    state_dict_pre_hooks: Dict[int, Callable] = dict(mod._state_dict_pre_hooks)
+    mod._state_dict_hooks.clear()
+    mod._state_dict_pre_hooks.clear()
+    try:
+        yield
+    finally:
+        mod._state_dict_hooks = state_dict_hooks
+        mod._state_dict_pre_hooks = state_dict_pre_hooks
+
+
+def exported_program_state_dict(mod: torch.nn.Module):
+    """
+    Given a model, returns a state_dict that's consistent with exported program's
+    state_dict.
+    """
+    with _disabled_load_state_dict_hooks(mod):
+        return mod.state_dict()
```

Comment:
Hmm what's the intended usage for this function?

Reply:
> Hmm what's the intended usage for this function?

This this for weight swapping. 

At time T1: you're running export, getting a exported program
At time T2: Some serving service serves an artifact from the exported program
At time T3: there is a recurring training job that just finished and updates the model state that is stored.
At time T4: the serving service is going to pick up the same compiled artifact with the new state that was just updated.

This is used to store the new model state at time T3.

something like, 
```
ep = export(model)
d = exported_program_state_dict(model)
# update ep's state_dict with d.
```